switchapi/CMakeLists.txt

```diff
@@ -60,15 +60,16 @@ add_library(switchapi_o OBJECT
 )
 
 target_include_directories(switchapi_o PRIVATE
-    ${SDE_INSTALL_PREFIX}/include
-    ${SDE_INSTALL_PREFIX}/include/target-utils # id
-    ${SDE_INSTALL_PREFIX}/include/target-utils/third-party
-    ${SDE_INSTALL_PREFIX}/include/target-utils/third-party/tommyds
-    ${SDE_INSTALL_PREFIX}/include/target-sys
+    ${SDE_INSTALL_DIR}/include
+    ${SDE_INSTALL_DIR}/include/bf_types
+    ${SDE_INSTALL_DIR}/include/target-utils # id
+    ${SDE_INSTALL_DIR}/include/target-utils/third-party
+    ${SDE_INSTALL_DIR}/include/target-utils/third-party/tommyds
+    ${SDE_INSTALL_DIR}/include/target-sys
 )
 
-find_library(TDI tdi ${SDE_INSTALL_PREFIX}/lib)
-find_library(TDI_JSON_PARSER tdi_json_parser ${SDE_INSTALL_PREFIX}/lib)
+find_library(TDI tdi ${SDE_INSTALL_DIR}/lib)
```

Comment:
Is TDI thread-safe? We have two different threads making calls to it.

This gives us two places in the code that independently establish linkages to the TDI libraries in the SDE. Each embeds some knowledge of how the linkage works. This is something that could conceivable vary between SDEs for different targets or between different versions of SDEs.  (I believe the conventions have already changed for newer versions of the Tofino SDE.)

For the longer term, we should attempt to abstract and encapsulate the interface to reduce the impact on CMake scripts when this kind of change takes place. We should also embed protections in the abstract interface or adopt appropriate usage conventions to ensure that two different subprojects may link to and use the SDE without interfering with each other.

Reply:
Yes TDI is thread safe. In old P4-OVS architecture we had TDI API's invoked through C++ interface via P4Runtime and also involved through krnlmon